src/api/dexPriceEstimator/DexPriceEstimatorApi.ts

```diff
@@ -1,7 +1,7 @@
 import BigNumber from 'bignumber.js'
 import { assert, TEN_BIG_NUMBER, ONE_BIG_NUMBER } from '@gnosis.pm/dex-js'
-import { Network } from 'types'
 import { ORDER_BOOK_HOPS_DEFAULT, ORDER_BOOK_HOPS_MAX } from 'const'
+import { DexPriceEstimatorConfig } from 'types/config'
```

Comment:
I think the config for the API should be fully defined in this file. 
In my opinion it should be the `Params`.

Also, note that this `Params` is the config for an specific implementation, maybe is time to move it out of the interface file and into something more like `RestPriceEstimationApi` or whatever we think it's convenient.

The factory responsibility would be to see the implementation the user wants to use (`type` for know implementation, `factory` for custom implementation --> probably too much for this PR)

Depending on the implementation, it would merge the defaults with the provider config, then it would hand all those parameters to the constructor of the implementation.

Reply:
> I think the config for the API should be fully defined in this file.
> In my opinion it should be the `Params`.

You mean, to have instead of a `config` object, to expand the config into the params themselves sort of like it was before?
 
> Also, note that this `Params` is the config for an specific implementation, maybe is time to move it out of the interface file and into something more like `RestPriceEstimationApi` or whatever we think it's convenient.
> 
> The factory responsibility would be to see the implementation the user wants to use (`type` for know implementation, `factory` for custom implementation --> probably too much for this PR)

Don't fully get this part.

> Depending on the implementation, it would merge the defaults with the provider config, then it would hand all those parameters to the constructor of the implementation.

That's one thing I'd like to avoid. In my view, the config provider should take care of using/overwriting defaults. Then we don't need any extra logic on the classes to do this merge.

Reply:
Mmm, nup, I think we should try to see it in a call or sth. Or I can try to do a counter PR, cause it's not easy to explain

Right now I'm super tired, I'll do a tiny attempt to do a PR, if I don't manage to do that now in literally 10min, I'll send it tomorrow

Reply:
We better then book a call for when you are available.